### PR TITLE
Added multiple keyboard button positions and mobile menu text string with multiple positions

### PR DESCRIPTION
**Pull request by ipkpjersi**
Technically this may not be necessary, and I can understand not wanting it, and I promise I won't be offended if you don't want this - but I think it's actually really good and worth having.

**Here are some examples of the menu text string positioning:**
Top:

Bottom:

**Here are some examples of the keyboard button positioning:**
Left: 

Top: 

 
Right:

### CODE REVIEW

**Comment by alarixnia** [comment]
FPS should be moved to the right in the second example to match legacy mudclients

**Comment by ipkpjersi** [comment]
It already is on the right in desktop mode fwiw, I'm not sure why it isn't in mobile mode.

**Comment by ipkpjersi** [comment]
I've not changed the FPS display for now as we could always change that in the future if we really needed to, but I did reposition the menu text when FPS is displaying versus not.

FPS displaying:
Screenshot_2024-05-09-12-59-07-81_40deb401b9ffe8e1df2f1cc5ba480b12


FPS not displaying:
Screenshot_2024-05-09-12-59-18-66_40deb401b9ffe8e1df2f1cc5ba480b12


I think this might be good to go as is now, as long as we do want these features I added. I've tested this PR pretty heavily and it seems to work perfectly for me.

File: src/ui/menu.c [hunk 1]
```diff
@@ -526,6 +526,8 @@ void mudclient_menu_item_click(mudclient *mud, int i) {
 }
 
 void mudclient_create_top_mouse_menu(mudclient *mud) {
+    mud->menu_text_position = "top"; /* TODO: extract this to a user-changeable option*/
```
Comment (alarixnia) [inline src/ui/menu.c:529]: This assignment makes the string read-only so it should be declared const.

Comment (alarixnia) [reply src/ui/menu.c]: Yeah it only matters since a string is an array (you can't write into "this"), menus are ints.

Comment (ipkpjersi) [reply src/ui/menu.c]: I'm not sure how to get it to work as a const then, I'm still quite inexperienced with C.

**Comment by misterhat** [comment]
> FPS should be moved to the right in the second example to match legacy mudclients

i think i moved it there since the UI was moved to the right, and FPS would often be obscured if a tab was open

**Pull request merged**